### PR TITLE
Send Slack Notification on Deployment

### LINKED ISSUE #77: Slack Integration
@mehrancodes I know that Slack integration is on your roadmap.  Is this something you're already working on?  If not, mind if I take it on?

I just didn't want to start if you had it in progress.

### PR DESCRIPTION
**Pull request by shawnhooper**
Fixes #77 

To test this:

1. Create a new Slack app.  Copy it's OAuth Access token.
2. Create a new GitHub secret with the OAuth Access token.
3. Invite the Slack app into the channel where you want to announce the deployments.
4. Add two new environment variables to the preview-provision.yml workflow:

          SLACK_ANNOUNCEMENT_ENABLED: true
          SLACK_BOT_USER_OAUTH_TOKEN: ${{ secrets.SLACK_BOT_USER_OAUTH_TOKEN }}
          SLACK_CHANNEL: "#deployments"  (replace this with the channel you invited the bot to)

5. Test a deployment.

The Slack Notification will only be sent when the site is first created.  Subsequent pushes to the same PR won't trigger the notification.

### CODE REVIEW

**Comment by shawnhooper** [comment]
image

This is proof of it working in my environment

File: app/Services/Forge/Pipeline/AnnounceSiteOnSlack.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+use Illuminate\Support\Facades\Notification;
+
+class AnnounceSiteOnSlack
+{
+    use Outputifier;
+
+    public function __invoke(ForgeService $service, Closure $next)
+    {
+        // End early if the slack bot token and channel are not set in the Forge service settings
+        if ( ! $service->setting->slackAnnouncementEnabled || ! $service->setting->slackBotUserOauthToken || ! $service->setting->slackChannel ) {
```
Comment (mehrancodes) [inline app/Services/Forge/Pipeline/AnnounceSiteOnSlack.php:31]: Do we need to check for the slack token and channel? As we are validating them in the ForgeSetting

**Review by mehrancodes — APPROVED** [review]
It looks great! 👨🏻‍💻 Thanks @shawnhooper for the great work! Everything works smoothly when I test it manually, too.

File: app/Services/Forge/Pipeline/AnnounceSiteOnSlack.php (hunk 1 above)
Comment (mehrancodes) [reply app/Services/Forge/Pipeline/AnnounceSiteOnSlack.php]: I tested it out and looks like the validator handles it 🚀. I would resolve the conversation

```bash
  FAIL    ---> The slack bot user oauth token field is required.
  FAIL    ---> The slack channel field is required.
```

**Comment by shawnhooper** [comment]
Thanks @mehrancodes !

**Pull request merged**